packages/kbn-router-to-openapispec/src/generate_oas.ts

```diff
@@ -68,15 +68,21 @@ export const generateOpenApiDocument = (
       },
     ],
     paths,
-    components: converter.getSchemaComponents(),
-    security: [
-      {
-        basicAuth: [],
-      },
-      {
-        apiKeyAuth: [],
+    components: {
+      ...converter.getSchemaComponents(),
+      securitySchemes: {
+        basicAuth: {
```

Comment:
Updated this based on complaints about this structure from https://editor.swagger.io/

FWIW I got a schema using `curl -uelastic:changeme http://localhost:5601/api/oas\?pathStartsWith\=/api/status | jq | pbcopy`